src/main/java/core/basesyntax/strategy/OperationStrategyImpl.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.strategy;
+
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlers;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation,
+            OperationHandler> operationHandlers) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    public OperationHandler getOperationHandler(FruitTransaction.Operation operation) {
+        if (operation == null) {
+            throw new NullPointerException();
```

Comment:
Manually throw NPE is a bad practice, so let's throw an IllegalArgumentException with informative message
```suggestion
            throw new IllegalArgumentException();
```